src/easylink/step.py

```diff
@@ -150,6 +144,32 @@ def validate_step(self, step_config: LayeredConfigTree) -> Dict[str, List[str]]:
             ]
         return errors
 
+    def get_implementation_node_name(self, step_config):
+        """Resolve a sensible unique node name for the implementation graph.
+        This method compares the step node names with the step names through the step hierarchy and
+        uses the full suffix of step names starting from wherever the two first differ. For example,
+        if we have node names step_3_loop_1 step_3_python_pandas the resulting implementation node name
+        will be step_3_loop_1_step_3_python_pandas. If all the node names and step names match, we have not
+        introduced any step degeneracies with e.g. loops or multiples, and we can simply use the implementation
+        name."""
+        step = self
+        node_names = []
+        step_names = []
+        while step:
+            node_names.append(step.name)
+            step_names.append(step.step_name)
+            step = step.parent_step
+
+        implementation_names = []
+        step_names.reverse()
```

Comment:
Is this all going to wreak havoc with parallel steps?

Reply:
I don't see why it would, but maybe you've caught something I didn't. My reasoning was that multiples will only have one parent step, as loops do, and like loops, the idea is that multiples will assign some disambiguated name to the paralell implementation nodes ("step_3_split_1" or something).